tests/test_res_template.cc

```diff
@@ -18,28 +18,62 @@
  * http://www.gnu.org/copyleft/gpl.html
  */
 
+#include <vector>
+#include <cstdio>
+#include <string>
+#include <iostream>
+#include <fstream>
+
 #include "catch.h"
 #include "../src/res_template.h"
 
 using namespace std;
 
 TEST_CASE("ResTemplate::FileMatrixToTemplate", "[res_template]") {
-  SECTION("Unreadable file") {
-    // REQUIRE_THROWS_AS();
+  string file = "matrix_file";
+  vector<string> matrix;
+  vector<string> matrix_comp;
+
+  remove(file.c_str());
+
+  ofstream output;
+  output.open(file);
+  output << " AC" << endl;
+  output << "A+ " << endl;
+  output << "C +" << endl;
+  output << endl;
+  output << "~AT" << endl;
+  output << "A +" << endl;
+  output << "T+ " << endl;
+  output.close();
+
+  ResTemplate res_template(file);
+  ResTemplate res_template_comp(file, true);
+
+  SECTION("Parse of forward matrix OK") {
```

Comment:
So in MatrixIO the matrix file is tested and exceptions thrown if file is not readable/existing or empty. I didn't include testing of the matrix format itself. If that is what you mean we should create an issue on more testing of matrix format in MatrixIO.